data/adapters/helpers.py

```diff
@@ -128,3 +128,151 @@ def to_float(str):
+    else:
+        raise(ValueError(
+            f'Loading data from file sets other than prediction sets and analysis sets is currently unsupported.'))
+    if 'publications' in file_set_object:
+        publication_ids.update(
+            get_publication_ids(portal_url, file_set_object))
+
+    # get samples metadata
+    samples = file_set_object.get('samples', [])
+    sample_ids = []
+    sample_term_ids = set()
+    donor_ids = set()
+    simple_sample_summaries = set()
+    treatment_ids = set()
+    for sample in samples:
+        sample_object = requests.get(
+            portal_url + sample + '/@@object?format=json').json()
+        sample_ids.append(sample_object.get('accession'))
+        if 'donors' in sample_object:
+            donors = sample_object.get('donors', [])
+            for donor in donors:
+                donor_object = requests.get(
+                    portal_url + donor + '/@@object?format=json').json()
+                donor_ids.add(donor_object.get('accession'))
+        if 'targeted_sample_term' in sample_object:
+            targeted_sample_term_object = requests.get(
+                portal_url + sample_object.get('targeted_sample_term') + '/@@object?format=json').json()
+            targeted_sample_term_name = targeted_sample_term_object.get(
+                'term_name')
+            classifications = ', '.join(sample_object.get('classifications'))
+            simple_sample_summary = f'{targeted_sample_term_name} {classifications}'
+            sample_term_ids.add(targeted_sample_term_object.get('term_id'))
+        else:
+            sample_terms = sample_object.get('sample_terms', [])
+            sample_term_names = set()
+            for sample_term in sample_terms:
+                sample_term_object = requests.get(
+                    portal_url + sample_term + '/@@object?format=json').json()
+                sample_term_names.add(sample_term_object.get('term_name'))
+                sample_term_ids.add(sample_term_object.get('term_id'))
+            sample_term_names = ', '.join(list(sample_term_names))
+            simple_sample_summary = f'{sample_term_names}'
+        if 'treatments' in sample_object:
+            treatment_term_names = set()
+            for treatment in sample_object.get('treatments', []):
+                treatment_object = requests.get(
+                    portal_url + treatment + '/@@object?format=json').json()
+                treatment_ids.add(treatment_object.get('term_id'))
+                treatment_term_names.add(
+                    treatment_object.get('treatment_term_name'))
+                if 'publications' in treatment_object:
+                    publication_ids.update(get_publication_ids(
+                        portal_url, treatment_object))
+            treatment_term_names = ', '.join(list(treatment_term_names))
+            simple_sample_summary = f'{simple_sample_summary} treated with {treatment_term_names}'
```

Comment:
trying to think if we want treatment_terms_name or summary instead, the latter contains duration info. The encode E2G annotations are split by treatment + duration, so would be useful to have that in sample summary, but might not be true for IGVF analysis sets?

Reply:
We do have duration, I can add that. The summary for treatments in IGVF also include amount & amount_units though.